ls1intum/Pyris
# Add endpoint for prometheus metrics

This PR adds a pull request for prometheus metrics. It exposes various predefined FastAPI and python metrics as well as a CPU usage and memory usage gauge.

**Notes to reviewers**: This PR relies on external library code and observed system behavior. That's why there are no tests as of right now. If anyone has an idea on which unit tests to create here, jsut request changes.

## Steps for testing
- Run the server and visit `/metrics`
- Verify you're seeing a page with various metrics.
  - Make sure _at least_ these metrics are present: `system_cpu_usage_percent`, `system_memory_usage_bytes`, `http_requests_total`, `http_request_size_bytes_sum`, `http_response_size_bytes_sum`.
- Reload the page and visit other endpoints.
- Check if the metrics have changed their values (i.e. if the CPU usage seems to be accurate or frozen)
